pkg/scheduler/framework/plugins/helper/spread_test.go

```diff
@@ -99,3 +101,159 @@ func TestGetPodServices(t *testing.T) {
 		})
 	}
 }
+
+func TestDefaultSelector(t *testing.T) {
+	const (
+		namespace                 = "test"
+		podName                   = "test-pod"
+		serviceName               = "test-service"
+		replicaSetName            = "test-replicaset"
+		replicationControllerName = "test-replicationcontroller"
+		statefulSetName           = "test-statefulset"
+
+		podLabelKey = "podLabelKey"
+		podLabelVal = "podLabelVal"
+
+		replicaSetLabelKey = "replicaSetLabelKey"
+		replicaSetLabelVal = "replicaSetLabelVal"
+
+		replicationLabelKey = "replicationLabelKey"
+		replicationLabelVal = "replicationLabelVal"
+
+		statefulSetLabelKey = "statefulSetLabelKey"
+		statefulSetLabelVal = "statefulSetLabelVal"
+	)
+
+	fakeInformerFactory := informers.NewSharedInformerFactory(&fake.Clientset{}, 0*time.Second)
+
+	// Create fake service
+	service := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: namespace,
+		},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{
+				podLabelKey: podLabelVal,
+			},
+		},
+	}
+	err := fakeInformerFactory.Core().V1().Services().Informer().GetStore().Add(service)
+	if err != nil {
+		t.Fatalf("Failed to add service to store: %v", err)
+	}
+
+	// Create fake ReplicaSet
+	addFakeReplicaSet := func() error {
+		replicaSet := &appsv1.ReplicaSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      replicaSetName,
+				Namespace: namespace,
+			},
+			Spec: appsv1.ReplicaSetSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						replicaSetLabelKey: replicaSetLabelVal,
+					},
+				},
+			},
+		}
+		return fakeInformerFactory.Apps().V1().ReplicaSets().Informer().GetStore().Add(replicaSet)
+	}
+
+	// Create fake ReplicationController
+	addFakeReplicationController := func() error {
+		replicationController := &v1.ReplicationController{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      replicationControllerName,
+				Namespace: namespace,
+			},
+			Spec: v1.ReplicationControllerSpec{
+				Selector: map[string]string{
+					replicationLabelKey: replicationLabelVal,
+				},
+			},
+		}
+		return fakeInformerFactory.Core().V1().ReplicationControllers().Informer().GetStore().Add(replicationController)
+	}
+
+	// Create fake StatefulSet
+	addFakeStatefulSet := func() error {
+		statefulSet := &appsv1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      statefulSetName,
+				Namespace: namespace,
+			},
+			Spec: appsv1.StatefulSetSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						statefulSetLabelKey: statefulSetLabelVal,
+					},
+				},
+			},
+		}
+		return fakeInformerFactory.Apps().V1().StatefulSets().Informer().GetStore().Add(statefulSet)
+	}
+
+	tests := []struct {
```

Comment:
also need one more testcase to cover the default case.

Reply:
I added a default case
```go
{
	name: "DefaultSelector for default case",
	pod: st.MakePod().Name(podName).
		Namespace(namespace).Label(podLabelKey, podLabelVal).
		Obj(),
	expect:          labels.Set{},
	addFakeService:  nil,
	addFakeResource: nil,
},
```